packages/typespec-python/src/http.ts

```diff
@@ -326,13 +326,20 @@ function emitHttpResponse(
     statusCodes: HttpStatusCodeRange | number | "*",
     response: SdkHttpResponse,
     method?: SdkServiceMethod<SdkHttpOperation>,
+    isException = false,
 ): Record<string, any> | undefined {
     if (!response) return undefined;
     let type = undefined;
-    if (response.type && !isAzureCoreModel(response.type)) {
+    if (isException) {
+        if (response.type && !isAzureCoreModel(response.type)) {
+            type = getType(context, response.type);
+        }
+    } else if (method && !method.kind.includes("basic")) {
+        if (method.response.type) {
+            type = getType(context, method.response.type);
+        }
+    } else if (response.type) {
```

Comment:
- For exception response, need to filter some AzureCoreModel (e.g. ErrorResponse)
- For normal response:
 1. lro/lropaging/paging API: there is always one kind of response type so use `method.response.type` directly
 2. basic API: there may be multi response types, so honor http response type.